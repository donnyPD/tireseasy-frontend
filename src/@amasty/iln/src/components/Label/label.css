.root {
  composes: font-normal from global;
  display: grid;
  gap: calc(0.5rem);
  --stroke: var(--venia-global-color-gray-600);
  align-items: center;
  color: rgb(var(--venia-global-color-gray-900));
  grid-template-columns: repeat(5, auto);
  justify-items: center;
  line-height: 1.5rem;
  cursor: pointer;
}

.root:hover .label {
  color: rgb(var(--venia-global-color-teal));
}

.root_selected {
  composes: root;
  color: rgb(var(--venia-brand-color-1-700));
}

.image {
  composes: imageSwatch from '../Swatch/swatch.css';
  min-width: 1.5rem;
  width: 1.5rem;
  height: 1.5rem;
  padding: 0;
}

.image_selected {
  composes: imageSwatch_selected from '../Swatch/swatch.css';
  min-width: 1.5rem;
  width: 1.5rem;
  height: 1.5rem;
}

.image_selected svg {
  width: 1rem;
  height: 1rem;
}

.checkbox {
  height: 1.5rem;
  pointer-events: none;
  width: 1.5rem;
  border-radius: 4px;
}

.checkbox svg {
  stroke: rgb(var(--stroke));
}

.root_selected .checkbox {
  --stroke: var(--venia-brand-color-1-700);
}

.checkbox ~ .label {
  color: rgb(var(--venia-global-color-text));
}

.root:active:enabled .checkbox,
.root:focus:enabled .checkbox {
  box-shadow: -3px 3px rgb(var(--venia-brand-color-1-100));
  outline: none;
}
