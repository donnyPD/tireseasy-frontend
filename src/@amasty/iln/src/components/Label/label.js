import React, { Fragment } from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { shape, string } from 'prop-types';

import defaultClasses from './label.css';
import Count from '../Count';
import MultiSelectIcon from './multiSelectIcon';
import ImageSwatch from '../Swatch/imageSwatch';
import { useIntl } from 'react-intl';

const Label = props => {
  const {
    isSelected,
    item: { label, image, count },
    onClick,
    showProductQuantities,
    isMultiSelect
  } = props;

  const { formatMessage } = useIntl();

  const classes = mergeClasses(defaultClasses, props.classes);
  const className = classes[`root${isSelected ? '_selected' : ''}`];

  const multiSelectIcon = isMultiSelect ? (
    <MultiSelectIcon isSelected={isSelected} />
  ) : null;

  const img = image ? (
    <ImageSwatch
      isSelected={isSelected}
      swatchData={image}
      classes={{
        imageSwatch: classes.image,
        imageSwatch_selected: classes.image_selected
      }}
    />
  ) : null;

  const labelText = typeof label === 'object' ? formatMessage(label) : label;

  return (
    <Fragment>
      <button
        className={className}
        title={label}
        onClick={onClick}
        type="button"
      >
        {multiSelectIcon}
        {img}
        <span className={classes.label}>{labelText}</span>
        <Count count={count} showProductQuantities={showProductQuantities} />
      </button>
    </Fragment>
  );
};

Label.propTypes = {
  classes: shape({
    root: string
  })
};

export default Label;
