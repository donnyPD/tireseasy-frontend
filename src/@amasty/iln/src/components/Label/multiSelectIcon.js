import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { bool, shape, string } from 'prop-types';

import defaultClasses from './label.css';
import { CheckSquare, Square } from 'react-feather';
const checkedIcon = <CheckSquare />;
const uncheckedIcon = <Square />;

const MultiSelectIcon = props => {
  const { isSelected } = props;
  const classes = mergeClasses(defaultClasses, props.classes);

  const icon = isSelected ? checkedIcon : uncheckedIcon;

  return <span className={classes.checkbox}>{icon}</span>;
};

MultiSelectIcon.propTypes = {
  isSelected: bool,
  classes: shape({
    root: string
  })
};

export default MultiSelectIcon;
