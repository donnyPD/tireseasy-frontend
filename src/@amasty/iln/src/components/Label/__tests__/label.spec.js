import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import Label from '../label';

jest.mock('../../Count', () => 'Count');
jest.mock('../multiSelectIcon', () => 'MultiSelectIcon');
jest.mock('../../Swatch/imageSwatch', () => 'ImageSwatch');
jest.mock('react-intl', () => ({
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));

test('render correct label', () => {
  const props = {
    item: {
      label: 'test',
      count: 4
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render label with image', () => {
  const props = {
    item: {
      label: 'test',
      count: 4,
      image: 'imgPatch'
    }
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render label with multiselect icon', () => {
  const props = {
    item: {
      label: 'test',
      count: 4
    },
    isMultiSelect: true
  };
  const tree = createTestInstance(<Label {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
