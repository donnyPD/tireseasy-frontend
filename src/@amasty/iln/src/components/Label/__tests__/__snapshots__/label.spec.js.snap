// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correct label 1`] = `
<button
  title="test"
  type="button"
>
  <span>
    test
  </span>
  <Count
    count={4}
  />
</button>
`;

exports[`render label with image 1`] = `
<button
  title="test"
  type="button"
>
  <ImageSwatch
    classes={
      Object {
        "imageSwatch": undefined,
        "imageSwatch_selected": undefined,
      }
    }
    swatchData="imgPatch"
  />
  <span>
    test
  </span>
  <Count
    count={4}
  />
</button>
`;

exports[`render label with multiselect icon 1`] = `
<button
  title="test"
  type="button"
>
  <MultiSelectIcon />
  <span>
    test
  </span>
  <Count
    count={4}
  />
</button>
`;
